### PR TITLE
[Origin Rewards] May rules implementation

### PR DESCRIPTION
**Pull request by franckc**
### Description:

Code changes to support Origin Rewards May rules:
 - added support for google attestation
 - added support for a new type of rule called ListingIdPurchased to buy a specific listing Id
 - defined the May rules config
 - refactor growth code to have a single file (under growth/campaigns) that contains the config for the rules. Previously the config was duplicated in several places.
 - added unit tests specific to May campaign for rules and adapter
 - updated marketplace handler from the listener to insert as many ListingPurchased/ListingSold rows as there were unit bought. This will allow us to support the behavior for the May campaign where if a user buys N units from a listing, they should get N times the reward.
 - changed the DApp campaign page to get extra data specific to the new rule for buying a listing Id: added fields listingId, titleKey and iconSrc. This will need some extra polishing when we implement the FE to display the May rules.

Note:
 - the copy I have used for the rule titles is just a placeholder. We'll need to get final copy from Matt.
 - iconSrc points to currently non-existent icons. We'll need to get those from Aure.

### Checklist:

- [X] Test your work and double-check to confirm that you didn't break anything
- [X] [Wrap any new text/strings with `fbt`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#wrapping-text) for translation
- [ ] If there are any new/changed strings to translate, [`npm run translate`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#integrating-translations)
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/docs)

### CODE REVIEW

**Comment by franckc** [comment]
@sparrowDom Just FYI in case you have any early comments. This is not ready yet for review...

**Comment by sparrowDom** [comment]
looks good so far to me, after a quick glance

**Comment by franckc** [comment]
@sparrowDom Still not ready for review as I'm still testing but FYI I added in the rule config a few new fields: iconSrc, titleText and listingId. Hopefully the front-end can use this data to show the rules for buying a specific listing. Let me know if you have any thoughts.

**Comment by franckc** [comment]
@sparrowDom This should now be ready for review...

**Comment by franckc** [comment]
@wanderingstan @tomlinton @nick Since Domen is going to be OOO for a few days, may I ask one of you for a review on this PR ? Sorry it is a bit long but most of it is unit tests. Happy to walk you through it if that helps... Thanks in advance !

**Comment by tomlinton** [comment]
Looks good from where I am sitting but I don't know much about this code 😞

File: dapps/marketplace/src/constants/Growth$FbtEnum.js [hunk 1]
```diff
@@ -7,10 +7,27 @@ const Growth$FbtEnum = {
...
+    'Earn 50 OGN per gift card purchased. Must be redeemed on Amazon US site.',
+  'growth.purchase.charity.title': 'Donate $100 to the GiveDirectly charity',
+  'growth.purchase.charity.details':
+    'Donation will be made by Origin on your behalf.',
+  'growth.purchase.housing.title': 'Book a home rental',
+  'growth.purchase.housing.details':
+    'Booking and stay must happen in the month of May.',
+  'growth.purchase.experience.title': 'TBD',
+  'growth.purchase.experience.details': 'TBD',
+  'growth.purchase.art.title': 'Purchase the XX by Trevor Jones',
```
Comment (nick) [inline dapps/marketplace/src/constants/Growth$FbtEnum.js]: need to update this?

Comment (franckc) [reply dapps/marketplace/src/constants/Growth$FbtEnum.js]: Yes, I'm waiting on final copies and listing Ids from Matt...

**Pull request merged**